nilearn/plotting/html_stat_map.py

```diff
@@ -1,148 +1,438 @@
+        Threshold a data array.
+    """
+    # Deal with automatic settings of plot parameters
+    if threshold == 'auto':
+        # Threshold epsilon below a percentile value, to be sure that some
+        # voxels pass the threshold
+        threshold = fast_abs_percentile(data) - 1e-5
+
+    # Threshold
+    threshold = float(threshold) if threshold is not None else None
+
+    # Mask data
+    if threshold is not None:
+        if threshold == 0:
+            data = np.ma.masked_equal(data, 0, copy=False)
+        else:
+            data = np.ma.masked_inside(data, -threshold, threshold, copy=False)
+    return data
+
+
+def _save_sprite(data, output_sprite, vmax, vmin, mask=None, cmap='Greys',
+                 format='png'):
+    """ For internal use.
+        Generate a sprite from a 3D Niimg-like object.
+    """
+
+    # Create sprite
+    sprite = _data2sprite(data)
+
+    # Mask the sprite
+    if mask is not None:
+        mask = _data2sprite(mask)
+        sprite = np.ma.array(sprite, mask=mask)
+
+    # Save the sprite
+    imsave(output_sprite, sprite, vmin=vmin, vmax=vmax, cmap=cmap,
+           format=format)
+
+    return sprite
+
+
+def _bytesIO_to_base64(file):
+    """ For internal use.
+        Encode a bytesIO virtual file as base64.
+        Also closes the file.
+    """
+    file.seek(0)
+    data = encodebytes(file.read()).decode('utf-8')
+    file.close()
+    return data
+
+
+def _save_cm(output_cmap, cmap, format='png'):
+    """ For internal use.
+        Save the colormap of an image as a png file.
+    """
+
+    # Hard-coded number of colors
+    n_colors = 256
+
+    # save the colormap
+    data = np.arange(0, n_colors) / (n_colors - 1)
+    data = data.reshape([1, n_colors])
+    imsave(output_cmap, data, cmap=cmap, format=format)
 
 
 class StatMapView(HTMLDocument):
     pass
 
 
-def _to_papaya(cmap, norm, abs_min, vmax, symmetric_cmap):
-    """Create colormaps ('lookup tables') given to papaya viewer"""
-    n_colors = 100
-    if not symmetric_cmap:
-        x = np.linspace(norm(abs_min), norm(vmax), n_colors)
-        rgb = cmap(x)[:, :3]
-        return {'positive_cmap': _rgb_to_papaya(x, rgb),
-                'negative_cmap': None}
-    papaya = {}
-    x = np.linspace(norm(-vmax), norm(-abs_min), n_colors)
-    y = np.linspace(0, 1, n_colors)
-    rgb = cmap(x)[::-1, :3]
-    papaya['negative_cmap'] = _rgb_to_papaya(y, rgb)
-    x = np.linspace(norm(abs_min), norm(vmax), n_colors)
-    rgb = cmap(x)[:, :3]
-    papaya['positive_cmap'] = _rgb_to_papaya(y, rgb)
-    return papaya
-
-
-def _rgb_to_papaya(x, cmap):
-    """Transform (values, rgb array) to format expected by papaya"""
-    cmap = [[float(m)] + list(map(float, col)) for m, col in zip(x, cmap)]
-    return cmap
-
-
-def view_stat_map(stat_map_img, threshold=None, bg_img='MNI152',
-                  vmax=None, cmap='cold_hot', symmetric_cmap=True):
+def _mask_stat_map(stat_map_img, threshold=None):
+    """ For internal use.
```

Comment:
If you are making functions private (leading underscore), it is not necessary to add that they should be for internal use.

Reply:
ok, thanks for the clarification. I got rid of all these lines.